### PR TITLE
add UT for aggregatestatus

### PR DESCRIPTION
**Pull request by Poor12**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add UT for aggregatestatus

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+
+	"github.com/stretchr/testify/assert"
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:16]: Move this line to above.

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+		oldObj                *unstructured.Unstructured
+		aggregatedStatusItems []workv1alpha2.AggregatedStatusItem
+		newObj                *unstructured.Unstructured
+		wantErr               error
+	}{
+		{
+			oldObj:                objServiceClusterIP,
+			aggregatedStatusItems: aggregatedStatusItemsNotLB,
+			newObj:                objServiceClusterIP,
+			wantErr:               nil,
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:101]: Without a test case name, the purpose of the test is not clear at a glance.

Comment (Poor12) [reply pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:106]: Thanks for advice. Add testcase name.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
lgtm

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+	oldDeploy := &appsv1.Deployment{}
+	newDeploy := &appsv1.Deployment{Status: appsv1.DeploymentStatus{Replicas: 2, ReadyReplicas: 2, UpdatedReplicas: 2, AvailableReplicas: 2, UnavailableReplicas: 0}}
+	oldObj, _ := helper.ToUnstructured(oldDeploy)
+	newObj, _ := helper.ToUnstructured(newDeploy)
+
+	tests := []struct {
+		name                  string
+		oldObj                *unstructured.Unstructured
+		aggregatedStatusItems []workv1alpha2.AggregatedStatusItem
+		newObj                *unstructured.Unstructured
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:41]: Does it makes better readability if we rename like?:
```diff
-               oldObj                *unstructured.Unstructured
+               curObj                *unstructured.Unstructured
                aggregatedStatusItems []workv1alpha2.AggregatedStatusItem
-               newObj                *unstructured.Unstructured
+               expectedObj           *unstructured.Unstructured
```

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+		wantErr               error
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:42]: Should we cover the case that expects an error?

If you don't want to cover the `abnormal` case, then the `wantErr` should not be there, which means all your test cases below do not allow `err`.

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go (hunk 3 above)
Comment (Poor12) [reply pkg/resourceinterpreter/defaultinterpreter/aggregatestatus_test.go:41]: seems better.

**Pull request merged**